app/models/patron_request.rb

```diff
@@ -550,7 +550,9 @@ def default_pickup_service_point
   # Returns default service point codes for all requests
   # @return [Array<String>]
   def default_pickup_service_points_codes
-    Folio::Types.service_points.where(is_default_pickup: true).map(&:code)
+    Folio::Types.service_points.where(is_default_pickup: true).reject do |service_point|
+      service_point.cant_pick_up.include?(patron.patron_group_name)
```

Comment:
I think this filtering belongs in `#pickup_destinations`, where we're already doing filtering for true visitors. The method (and its comment) suggest to me that it is supposed to get the default service points regardless of the request.

Reply:
Ok, I can try a PR with my idea. Maybe there's a reason you needed to introduce the logic here that I'm not getting.

Reply:
Thanks.